wsf_scraping/pipelines.py

```diff
@@ -122,13 +128,17 @@ def check_keywords(self, item, spider_name, base_pdf_path):
     def process_item(self, item, spider):
         """Process items sent by the spider."""
 
-        base_pdf_path = os.path.join('/tmp', item['pdf'])
+        base_pdf_path = os.path.join(tempfile.gettempdir(), item['pdf'])
         file_hash = get_file_hash(base_pdf_path)
         if self.database.is_scraped(file_hash):
             # File is already scraped in the database
             raise DropItem(
                 'Item footprint is already in the database'
             )
+        if not item['pdf']:
+            raise DropItem(
+                'Empty filename, could not parse the pdf.'
+            )
```

Comment:
It would be better to move this to the top of the function, so you can skip getting the file hash and the database call for a file that you can’t read.